docs/development/pigs/pig-008.rst

```diff
@@ -0,0 +1,594 @@
+
+We propose to implement a `.peek()` method on all datasets:
+
+.. code::
+
+    dataset.peek()
+
+In an interative environment this will show a default visualisation of the
+dataset including counts / data, IRFs, residuals etc.
+
+
+Simulation of `MapDataset` and `SpectrumDataset`
+------------------------------------------------
+The `MapDataset` and `SpectrumDataset` classes will be the central classes
+for map and spectrum based analyses. In many cases it's useful to simulate
+counts from a predicted number of counts model. The direct sampling of
+binned data for the datasets could be supported as following (illustrated
+for the `MapDataset`):
+
+.. code::
+
+    # allow counts=None on init
+    dataset = MapDataset(counts=None)
+
+    # to sample a counts map from npred
+    dataset.fake(random_seed=0)
+
+    fit = Fit(dataset)
+    fit.optimize()
+
+Which calls `np.random.poisson()` internally on the npred map and sets it
+as a `.counts` map on the dataset. This works equivalently for the
+`SpectrumDataset`, `SpectrumDatasetOnOff` and `MapDatasetOnOff`. The on-off
+dataset require a background model for this to work.
+
+Unbinned simulation (sampling of event lists) is addressed in a separate PIG.
+
+
+`Datasets`
+----------
+
+To combine multiple datasets in a joint-likelihood analysis we propose to introduce
+a `Datasets` class. Its responsibility is to add up the log-likelihood values of
+the individual datasets and join the parameter lists. The `Datasets` class
+also represents the main interface to the `Fit` class.
+
+The following example shows how a joint-likelihood analysis of multiple observations
+is supposed to work:
+
+.. code ::
+
+    from gammapy.utils.fitting import Datatsets
+
+    model = SourceModel.read()
+
+    map_datasets = []
+
+    for obs_id in [123, 124, ...]:
+        bkg_model = BackgroundModel.read("obs-{}/background.fits".format(obs_id))
+        dataset = MapDataset(model=model, background=bkg_model)
+        map_datasets.append(dataset)
+
+    datasets = Datasets(map_datasets)
+
+    # check whether all contained datasets are of the same type
+    datasets.is_all_same_type
+
+    datasets.stat()
+
+    fit = Fit(datasets)
+
+    # or maybe simpler
+
+    fit = Fit(map_datasets)
+    fit.optimize()
+
+
+The linking of parameters of the spectral model is natively achieved, as the same
+instance of the spectral model is passed to two different datasets, while the
+background model is created for every dataset separately.
+
+A joined spectral fit across multiple observations:
+
+.. code ::
+
+    model = SpectralModel()
+
+    spectrum_dataset_1 = SpectrumDataset(model, ...)
+    spectrum_dataset_2 = SpectrumDataset(model, ...)
+
+    datasets = Datasets([spectrum_dataset_1, spectrum_dataset_2])
+
+    fit = Fit(datasets)
+    fit.optimize()
+
+
+Combined spectral / flux points analysis:
+
+.. code ::
+
+    model = SpectralModel()
+
+    spectrum_dataset = SpectrumDataset(model=model)
```

Comment:
Here you could explicitly mention `stat="cash``, to make it visible in one example that the stats on datasets can be different.